File: medplum-java/medplum-model/src/main/java/com/fhir/DataRequirement_CodeFilter.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableDataRequirement_CodeFilter.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface DataRequirement_CodeFilter {
  Optional<String> searchParam();

  Optional<String> path();

  Optional<List<Extension>> extension();

  Optional<List<Coding>> code();

  Optional<String> id();

  Optional<List<Extension>> modifierExtension();

  Optional<canonical> valueSet();

  static ImmutableDataRequirement_CodeFilter.Builder builder() {
    return ImmutableDataRequirement_CodeFilter.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/DataRequirement_CodeFilter.ts
import { canonical } from "./canonical";
import { Coding } from "./Coding";
import { Extension } from "./Extension";

export interface DataRequirement_CodeFilter {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    path: string | null | undefined;
    searchParam: string | null | undefined;
    valueSet: canonical | null | undefined;
    code: Coding[] | null | undefined;
}
